### ISSUE TITLE
Move off pdfrw?

### ISSUE THREAD

**Issue by psolin** [psolin/pymkup#13]
The library does not appear to be updated anymore.

**Comment by supersonic1999** [comment]
https://github.com/maxpmaxp/pdfreader was updated 2 months ago, that project might more suitable

**Comment by psolin** [comment]
Testing this library out, it seems to work very well. I don't think it would be that difficult to integrate it. For example, this gets you a list of all markups on the first page:

```python
>>> file_name = 'markup-deep-spaces.pdf'
>>> fd = open(file_name, "rb")
>>> doc = PDFDocument(fd)
>>> catalog = doc.root
>>> catalog.Pages.Kids[0].Kids[0].Annots
```

I think this is a great choice. Even though we would lose the ability to develop ways to split the PDF or edit markups,  there's still a pretty big risk to keeping pdfrw that has 91 open issues. Reading through some of them, I'm not the first person to think about migrating to something else. I'll take a look at creating a new branch over the weekend to get this project moving along. Thanks!

**Comment by psolin** [comment]
I was able to make the library work, but my tests are at 212ms up from around 30ms. There's something that isn't optimized somewhere. I could try branching it if you wanted to take a look.

**Comment by supersonic1999** [comment]
Yea go for it.

**Comment by supersonic1999** [comment]
Been looking into it last few hours. Problem seems to be with both pdfreader import and colormap import

Looking down the log tree, it seems to be related to them both using pkg_resources as an import in their respective libraries.

Googling 'python pkg_resources SLOW' shows many people with the same issue.

**Comment by psolin** [comment]
Ok thank you for looking into it. I can probably switch shapely to matlabplot and see if it does anything for speed. From there, I can just keep redoing the code, maybe getting rid of try/except clauses. I'm hoping with Pillow that pdfreader will fix the PNG issue. When I incorporate it into a personal project, I can do more testing that way, but I don't have any more test files to upload. 

E: I can also probably replace color map.

**Comment by psolin** [comment]
I was able to replace colormap with matplotlib, but I'm seeing no difference in speed.

**Comment by supersonic1999** [comment]
The main issue with pdfreader and colormap Is they both import a library called pkg_resources.

 If any used library, imports that library it will slow the initial load down by that amount. Only way to fix it is to remove all packages that use that import or get them to fix it, which in this case is pdfreader and color map.

Also it depends if that slowdown is relevant to the project or not.

**Comment by psolin** [comment]
I was able to replace colormap and shapely with matplotlib, but that also uses pkg_resources, unfortunately. I removed a few try/except clauses because pdfreader can handle Nonetypes better, and gained about 10ms or so in the test.

In all honesty, I don't care too much about slowdown versus stability, but I'll keep trying to chip away at the code to make it faster anyway. I am looking to merge the pdfreader branch with main some time over the next week.
